atc/exec/task_step.go

```diff
@@ -181,6 +177,22 @@ func (step *TaskStep) Run(ctx context.Context, state RunState) error {
 		config.Limits.Memory = step.defaultLimits.Memory
 	}
 
+	events := make(chan runtime.Event, 1)
+	go func(logger lager.Logger, config atc.TaskConfig, events chan runtime.Event, delegate TaskDelegate) {
+		for ev := range events {
+			switch ev.EventType {
+			case runtime.InitializingEvent:
+				step.delegate.Initializing(logger, config)
+
+			case runtime.StartingEvent:
+				step.delegate.Starting(logger, config)
+
+			case runtime.FinishedEvent:
```

Comment:
Doesn't look like this event is called ?

Reply:
we'll address these changes in a follow up PR because it's been a while since this PR has been open and keeping it up to date with master is problematic.